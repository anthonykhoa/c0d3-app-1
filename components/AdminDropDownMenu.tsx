import React, { useState } from 'react'
import _ from 'lodash'
import '../scss/navbar.scss'
import NavLink from './NavLink'

const adminStyle = {
  padding: 'none',
  margin: '0 0 0 0'
}

const AdminDropDownMenu: React.FC = () => {
  const [showOptions, setShowOptions] = useState('dropdown-menu')
  const showIt = () => setShowOptions('dropdown-menu show')
  const stopIt = () => setShowOptions('dropdown-menu')
  return (
    <div className="dropdown">
      <div
        style={adminStyle}
        data-testid="adminButton"
        onClick={showIt}
        onMouseOut={stopIt}
        className="btn btn-secondary dropdown-toggle"
        aria-expanded="false"
      >
        Admin
      </div>
      <div
        data-testid="menu"
        style={adminStyle}
        className={showOptions}
        onMouseOver={showIt}
        onMouseOut={stopIt}
      >
        <NavLink path="/admin/lessons" className="dropdown-item">
          Lesson
        </NavLink>
        <NavLink path="/admin/lessons" className="dropdown-item">
          Users
        </NavLink>
        <NavLink path="/admin/lessons" className="dropdown-item">
          Alerts
        </NavLink>
      </div>
    </div>
  )
}

export { AdminDropDownMenu }
