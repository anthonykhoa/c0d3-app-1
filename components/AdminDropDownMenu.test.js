import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { AdminDropDownMenu } from './AdminDropDownMenu'

describe('MdInput Component', () => {
  test('Should add `show` class when Admin div is clicked on', () => {
    const { container, queryByTestId } = render(<AdminDropDownMenu />)
    const adminButton = queryByTestId('adminButton')
    const menu = queryByTestId('menu')

    fireEvent.click(adminButton)

    expect(menu.classList.contains('show')).toBe(true)
    expect(container).toMatchSnapshot()
  })

  test('Should remove `show` class onMouseOut of menu', () => {
    const { container, queryByTestId } = render(<AdminDropDownMenu />)
    const menu = queryByTestId('menu')

    fireEvent.mouseOut(menu)
    expect(menu.classList.contains('show')).toBeFalsy
    expect(container).toMatchSnapshot()
  })
})
