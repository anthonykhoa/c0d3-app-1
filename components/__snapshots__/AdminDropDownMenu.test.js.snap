// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin dropdown menu component Should add \`show\` class when Admin div is clicked on 1`] = `
<div>
  <div
    class="dropdown"
  >
    <div
      aria-expanded="false"
      class="btn btn-secondary dropdown-toggle"
      data-testid="adminButton"
      style="margin: 0px 0px 0px 0px;"
    >
      Admin
    </div>
    <div
      class="dropdown-menu show"
      data-testid="menu"
      style="margin: 0px 0px 0px 0px;"
    >
      <a
        class="dropdown-item"
        href="/admin/lessons"
      >
        Lesson
      </a>
      <a
        class="dropdown-item"
        href="/admin/users"
      >
        Users
      </a>
      <a
        class="dropdown-item"
        href="/admin/alerts"
      >
        Alerts
      </a>
    </div>
  </div>
</div>
`;

exports[`Admin dropdown menu component Should remove \`show\` class onMouseOut of menu 1`] = `
<div>
  <div
    class="dropdown"
  >
    <div
      aria-expanded="false"
      class="btn btn-secondary dropdown-toggle"
      data-testid="adminButton"
      style="margin: 0px 0px 0px 0px;"
    >
      Admin
    </div>
    <div
      class="dropdown-menu"
      data-testid="menu"
      style="margin: 0px 0px 0px 0px;"
    >
      <a
        class="dropdown-item"
        href="/admin/lessons"
      >
        Lesson
      </a>
      <a
        class="dropdown-item"
        href="/admin/users"
      >
        Users
      </a>
      <a
        class="dropdown-item"
        href="/admin/alerts"
      >
        Alerts
      </a>
    </div>
  </div>
</div>
`;
